qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+
+    def on_off_status(self) -> str:
+        """
+        tells motor on off status
+        """
+        val = self.ask(f"MG _MO{self._axis}")
+        if val[0] == "1":
+            return "off"
+        else:
+            return "on"
+
+    def servo_here(self) -> None:
+        """
+        servo at the motor
+        """
+        self.write(f"SH {self._axis}")
+
+    def begin(self) -> None:
+        """
+        begins motion of the motor and waits until motor stops moving
+        """
+        self.write(f"BG {self._axis}")
+
+    def wait_till_motor_motion_complete(self) -> None:
+        """
+        this method on call waits for motion to complete on the motor
+        """
+        try:
+            while int(float(self.ask(f"MG _BG{self._axis}"))):
+                pass
+        except KeyboardInterrupt:
+            self.root_instrument.abort()
+            self.off()
+
+    def error_magnitude(self) -> float:
+        """
+        gives the magnitude of error, in drive step counts, for axes
+
+        a step count is directly proportional to the micro-stepping
+        resolution of the stepper drive.
+        """
+        return float(self.ask(f"QS{self._axis}=?"))
+
+
+class DMC4133Controller(GalilMotionController):
+    """
+    Driver for Galil DMC-4133 Controller
+    """
+
+    def __init__(self, name: str, address: str, **kwargs: Any) -> None:
+        """
+        Initializes the DMC4133Controller class
+
+        Args:
+            name: name for the instance
+            address: address of the controller burned in
+        """
+        super().__init__(name=name, address=address, **kwargs)
+
+        self.add_parameter(
+            "position_format_decimals",
+            get_cmd=None,
+            set_cmd="PF 10.{}",
+            vals=Ints(0, 4),
+            docstring="sets number of decimals in the format "
+                      "of the position",
+        )
+
+        self.add_parameter(
+            "absolute_position",
+            get_cmd=self._get_absolute_position,
+            set_cmd=None,
+            unit="quadrature counts",
+            docstring="gets absolute position of the motors "
+                      "from the set origin",
+        )
+
+        self.add_parameter(
+            "wait",
+            get_cmd=None,
+            set_cmd="WT {}",
+            unit="ms",
+            vals=Multiples(min_value=2, max_value=2147483646, divisor=2),
+            docstring="controller will wait for the amount of "
+            "time specified before executing the next "
+            "command",
+        )
+
+        self._set_default_update_time()
+        self.add_submodule("motor_a", Motor(self, "A"))
+        self.add_submodule("motor_b", Motor(self, "B"))
+        self.add_submodule("motor_c", Motor(self, "C"))
+        self.add_submodule("plane_ab", VectorMode(self, "AB"))
+        self.add_submodule("plane_bc", VectorMode(self, "BC"))
+        self.add_submodule("plane_ac", VectorMode(self, "AC"))
```

Comment:
this assumes that 3 motors are connected to the controller, right? Is there a way we can ask controller what motors does it have? that would be best. otherwise, perhaps we need an argument to the constructor to let the user specify what motors are available (as a list, e.g. `['a','b','c']`) and then generate the submodules only for those.

that would also affect the methods like `_get_absolute_position`.

additionally, perhaps it makes sense to make a channel list out of motors.

Reply:
ah, i see, so the model of the controller defines the number of motors attached, i see.

>  we can read the IDN and get the number of motors connected from the message as if 4 motors are connected the message would read that it is a DMC4143 controller

i like this idea! but if it's too much of an extra effort, then let's not do it in this PR.

Reply:
Ok, then will leave it for another PR and keep the current implementation